### ISSUE TITLE
Loss function per output layer activation function

### ISSUE THREAD

**Issue by aawadall** [aawadall/Deep-Learning#4]
Make output layer activation function impact how loss function is calculated
in case of softmax it behaves different than sigmoid
